token/core/identity/tms/wm.go

```diff
@@ -447,8 +447,8 @@ func (lm *LocalMembership) addResolver(Name string, Type string, EnrollmentID st
 }
 
 func (lm *LocalMembership) getAnonymousResolver(label string) *Resolver {
-	lm.resolversMutex.RLock()
-	defer lm.resolversMutex.RUnlock()
+	lm.resolversMutex.Lock()
```

Comment:
but here read lock should be enough because we are not modifying the data structues.

Reply:
👍

Reply:
@adecaro : fixed ;)